java/src/Ice/src/main/java/com/zeroc/Ice/InputStream.java

```diff
@@ -2974,14 +2966,47 @@ private void initEncaps() {
 
     private void traceSkipSlice(String typeId, SliceType sliceType) {
         if (_traceSlicing && _logger != null) {
-            com.zeroc.IceInternal.TraceUtil.traceSlicing(
+            TraceUtil.traceSlicing(
                     sliceType == SliceType.ExceptionSlice ? "exception" : "object",
                     typeId,
                     "Slicing",
                     _logger);
         }
     }
 
+    /**
+     * We need to override the resolveClass method of ObjectInputStream so that we can use the same
+     * class-lookup mechanism as elsewhere in the Ice run time.
+     */
+    private class ObjectInputStreamWrapper extends java.io.ObjectInputStream {
+        public ObjectInputStreamWrapper(Instance instance, java.io.InputStream in)
+                throws IOException {
+            super(in);
+            _instance = instance;
+        }
```

Comment:
Both `Ice` and `IceInternal` had classes named `ObjectInputStream`... So I couldn't just move the file over.
Thankfully, the `IceInternal` version was only used in a single place, right here.

So, the easiest fix was to just 'inline' the class by copying it into a file-scoped private class instead.
But, I still went ahead and renamed it to `ObjectInputStreamWrapper`, to make it easier to disambiguate the two in the future.